docs/specs/src/hwp_3_correctness.md

```diff
@@ -0,0 +1,404 @@
+  Holochain provides the affordances of validation and warranting.
+
+3.  Protection from Attack categories
+
+4.  Resilience to Human Error
+
+### Gating Access via Cryptographic Object Capabilities
+
+To use a Holochain application end-users must trigger zome calls that
+affect local state changes on their Source Chains. Additionally zome
+functions can make calls to other zome functions on remote nodes in the
+same app, or to other DNAs running on the same conductor. All of these
+calls must happen in the context of some kind of permissioning system.
+Holochain\'s security model for calls is based on the
+Object-capability[^7] security model, but for a distributed
+cryptographic context in which we use cryptographic signatures to prove
+the necessary agency for taking action.
+
+Access is thus mediated by Capability grants of 4 types:
+
+-   author: only the agent owning the source change can make the zome
+  call
+
+-   assigned: only the named agent(s) can make the zome call
+
+-   transferrable: anybody with the given token can make the zome call
+
+-   unrestricted: anybody can make the zome call (no secret nor proof of
+  authorized key needed to use this capability)
+
+All zome calls must be signed and also take a required capability claim
+parameter that MUST be checked by the system for making the call. Agents
+record capability grants on their source chains and distribute them as
+necessary according to the application\'s needs. Receivers of grants can
+record them as claims (usually as a private entry) on their chains for
+later lookup. The \"agent\" type grant is just the agent\'s public key.
+
+### Validation & Warranting
+
+We have already covered how Holochain's agent-centric validation and
+intrinsic data integrity provides security from malicious actors trying
+to introduce invalid or incorrect information into an Application's
+network as every agent can deterministically verify data and thus
+self-secure. It is also important, however, to be able to eject
+malicious actors who generate or propagate such invalid data from
+network participation so as to secure against the resource drain that
+such action could incur.
+
+More details on warranting \[Todo: ACB\] \[expand on: one good apple
+heals the bunch\]
+
+### Security from Attack Categories
+
+#### Consensus​ ​Attacks
+
+This whole category of attack starts from the assumption that consensus
+is required for distributed systems. Because Holochain doesn't start
+from that assumption the attack category really doesn't apply, but it's
+worth mentioning because there​ ​are​ ​a​ ​number​ ​of​ ​attacks​ ​on​
+​blockchain​ ​which​ ​disrupt their distributed computing solution by
+having collusion between the majority of the nodes. ​The​ ​usual
+thinking​ ​is​ ​that​ ​it​ ​takes​ ​a​ ​large​ ​number​ ​of​ ​nodes45​
+​and​ ​massive​ ​amounts​ ​of​ ​computing​ ​power or financial
+incentives46​ ​to prevent​ ​undue​ ​hijacking​ ​of​ ​consensus.​
```

Comment:
```suggestion
incentives​ ​to prevent​ ​undue​ ​hijacking​ ​of​ ​consensus.​
```

Reply:
I see some other random numbers scattered around, please check for them in general